src/Connection/PhpiredisSocketConnection.php

```diff
@@ -308,7 +310,10 @@ public function connect()
     {
         if (parent::connect() && $this->initCommands) {
             foreach ($this->initCommands as $command) {
-                $this->executeCommand($command);
+                $response = $this->executeCommand($command);
+                if ($response instanceof ResponseInterface && $response instanceof ErrorResponseInterface) {
+                    $this->onConnectionError($command->parseResponse($response), 0);
```

Comment:
`$response->getMessage()` is enough, anyway don't worry I'll make these changes myself (and add a couple of test cases).